### PR TITLE
Move the value's serialization from Graph into Storage

### PR DESCRIPTION
**Pull request by darionyaphet**
* Move the value's serialization from `graph layer` into `storage layer`.

* Add field name

They are used to judge index operation.

### CODE REVIEW

File: src/graph/InsertEdgeExecutor.cpp [hunk 1]
```diff
@@ -47,51 +48,59 @@ void InsertEdgeExecutor::execute() {
...
         }
+
         {
             auto &out = edges[index++];
             out.key.set_src(src);
             out.key.set_dst(dst);
             out.key.set_ranking(rank);
             out.key.set_edge_type(edgeType_);
-            out.props = writer.encode();
+            for (auto property : properties_) {
```
Comment (dutor) [inline src/graph/InsertEdgeExecutor.cpp]: Serialize the property names for every row of values?

I seriously don't agree with that.

Comment (darionyaphet) [reply src/graph/InsertEdgeExecutor.cpp]: In `Storage Layer`, I wish to check the fields is indexed or not .

**Comment by dangleptr** [comment]
After talked with @sherman-the-tank,  we'd better keep the serialization at query engine.  
There are several reasons:
1.  Performance,   thrift encoding/decoding is much slower than dataman.
2. In our case, we only need to extract some column value from encoded value.  And dataman is designed for it,  it support read one column without decode the whole value. 

@darionyaphet

**Comment by darionyaphet** [comment]
`dataman` support read columns to avoid deserialize the whole binary value. This is a good point. :)

As a result, should just support fields in storage layer to make sure I can pick out the fields .

**Pull request closed without merging**